### PR TITLE
Use New WP_ENVIRONMENT_TYPE Constant for Site Environment

### PR DESCRIPTION
**Pull request by tarecord**
## What does this do/fix?

[SQONE-456](https://moderntribe.atlassian.net/browse/SQONE-456?atlOrigin=eyJpIjoiOGI2Nzk4ZTAzNTMyNDdlZThlNTc0ZmIzNjI2MzJhMGMiLCJwIjoiaiJ9)

Updates the method for checking the current site environment to use the `WP_ENVIRONMENT_TYPE` constant [added in Core v5.5](https://make.wordpress.org/core/2020/07/24/new-wp_get_environment_type-function-in-wordpress-5-5/).

## Tests

Does this have tests?

- [ ] Yes
- [x] No, this doesn't need tests because...it's just aligning currently available functionality.
- [ ] No, I need help figuring out how to write the tests.

### CODE REVIEW

**Review by defunctl — CHANGES_REQUESTED** [review]
Also check the `.github/workflows` folder that references the old defines as well as in the README.md

File: wp-config-environment.php [hunk 1]
```diff
@@ -155,7 +155,7 @@ function tribe_getenv( $name, $default = null ) {
 // Assign default constant value overrides for production
 // ==============================================================
 
-if ( defined( 'ENVIRONMENT' ) && ENVIRONMENT === 'PRODUCTION' ) {
+if ( defined( 'WP_ENVIRONMENT_TYPE' ) && WP_ENVIRONMENT_TYPE === 'production' ) {
```
[the remark this answers is not in the document]
Comment (tarecord) [reply wp-config-environment.php:163]: Good insight, I've updated the references.

**Comment by tarecord** [comment]
Good insight, references in `.github/workflows` have been updated.

**Comment by nickpelton** [comment]
Looks like a good conversion. Only note, we may want to set the new key and the old key JIC any of our external tools, like TribeLibs references the old global as well.

**Comment by tarecord** [comment]
Does the `local-config.php` ever make it to the server, or is it only for local environments? I was thinking we could define them there as defaults:

```php
/**
 * Set the current environment type. Accepted values:
 * - production (default)
 * - staging
 * - development
 * - local
 *
 * @link https://make.wordpress.org/core/2020/07/24/new-wp_get_environment_type-function-in-wordpress-5-5/
 */
define( 'WP_ENVIRONMENT_TYPE', 'development' );

// Fallback for older environments.
define ( 'ENVIRONMENT', 'DEV' );
```

**Comment by nickpelton** [comment]
It depends on who the host because we don't know how they treat the wp-config.php. In general we either use the .env, local-config.php, or other environment var setting tools. I'd say 70% of the time, we generate a local-config.php. If you look in the GHA deploy script you'll see we generate one on a deploy.

**Comment by nickpelton** [comment]
A link to where we create the local-config.php on a deploy https://github.com/moderntribe/square-one/blob/main/.github/workflows/deploy-dev.yml#L65

**Pull request merged**